sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc

```diff
@@ -615,18 +615,18 @@ global variable is allocated for each device, and that instance is shared by
 all kernels that belong to the same context and are submitted to the same device,
 regardless of which _device image_ contains the kernel.  When this property is
 specified, it is an assertion by the user that the device global is referenced
-only from kernels that are contained by the same _device image_.  An
+only from kernels that are contained by the same _device image_ instance, a _device image_ loaded onto a specific device. An
 implementation may be able to optimize accesses to the device global when this
 property is specified (especially on an FPGA device), but the user must be aware
-of which _device image_ contains the kernels that use the variable.
+of which _device image_ instance contains the kernels that use the variable.
 
 A device global that is decorated with this property may not be accessed from
-kernels that reside in different _device images_, either by direct reference
+kernels that reside in different _device images_ instances, either by direct reference
 to the variable or indirectly by passing the variable's address to another
 kernel.  The implementation is required to diagnose an error if the kernels
-that directly access a variable do not all reside in the same _device image_,
+that directly access a variable do not all reside in the same _device image_ instance,
 however no diagnostic is required for an indirect access from another _device
-image_.
+image_ instance.
 
 When a device global is decorated with this property, the implementation
 re-initializes it whenever the _device image_ is loaded onto the device.  As a
```

Comment:
From our offline discussion, we could go with something like this to be more useful than just saying "implementation defined" while still providing some implementation flexibility:

> When a device global is decorated with this property, in addition to the general device_global rules about initialization mentioned previously, the implementation may also re-initialize it at implementation-defined times.  However, the implementation is guaranteed not to re-initialize the device_global on a given deveice if all kernels run and all copied to/from device_image_scope device_globals on that device are from the same device image.

Reply:
How about this:

> A device global variable is guaranteed to be initialized for a device prior to the first time it is accessed (whether from a kernel or a copy operation).  Device globals may also be re-initialized at implementation-defined times if multiple device images are used on the same device.  To avoid unexpected re-initializations, applications should ensure that all kernels that are enqueued to a device _D_ come from the same device image.  In addition, applications should ensure that all device global copy operation enqueued to device _D_ correspond to that same device image.

Reply:
I like @gmlueck  version better.

Reply:
I like that.